hugr-core/src/extension/prelude.rs

```diff
@@ -255,8 +247,92 @@ pub const ERROR_TYPE: Type = Type::new_extension(ERROR_CUSTOM_TYPE);
 pub const ERROR_TYPE_NAME: TypeName = TypeName::new_inline("error");
 
 /// Return a Sum type with the first variant as the given type and the second an Error.
-pub fn sum_with_error(ty: Type) -> SumType {
-    SumType::new([ty, ERROR_TYPE])
+pub fn sum_with_error(ty: impl Into<TypeRowRV>) -> SumType {
+    result_type(ty, ERROR_TYPE)
+}
+
+/// An optional type, i.e. a Sum type with the first variant as the given type and the second as an empty tuple.
+#[inline]
+pub fn option_type(ty: impl Into<TypeRowRV>) -> SumType {
+    result_type(ty, TypeRow::new())
+}
+
+/// A result type, i.e. a two-element Sum type where the first variant
+/// represents the "Ok" value, and the second is the "Error" value.
+#[inline]
+pub fn result_type(ty_ok: impl Into<TypeRowRV>, ty_err: impl Into<TypeRowRV>) -> SumType {
```

Comment:
it seems more natural to call this `either` rather than result, result things could wrap this when convenient?